.github/workflows/deploy-docker-compose.yml

```diff
@@ -0,0 +1,161 @@
+name: Deploy Docker Compose
+
+on:
+  workflow_call:
+    inputs:
+      environment:
+        type: string
+        description: "The deployment environment (e.g., production, staging)"
+        required: true
+      docker-compose-file:
+        type: string
+        default: "./docker-compose.yml"
+        description: "Path to the Docker Compose file (Default: ./docker-compose.yml)"
+      main-image-name:
+        type: string
+        description: "The name of the main image for checking if it exists with the given tag"
+        required: true
+      image-tag:
+        type: string
+        description: "Image tag to deploy (default: pr-<number> if PR exists, latest for default branch)"
```

Comment:
Also most of our ls1 repositories uses monorepo where they have **at least** `client` & `server`. This `main-image-name` & `image-tag` wouldn't be a problem since most of them are using environment variables and use them inside the docker compose (image names/versions) but if we keep this inputs as it is then most of the repos will use one of the images name just to be able to use this reusable workflow